### PR TITLE
Payment update

### PR DESCRIPTION
**Pull request by NeilAllavarpu**
Slightly rework payment view, add stripe integration

### CODE REVIEW

File: src/components/FeatureViews/Registration/RegistrationCart/RegistrationCartContainer.js [hunk 1]
```diff
@@ -67,284 +67,285 @@ const GET_COURSES_AND_STUDENTS_TO_REGISTER = gql`
...
+                                        student={studentData.find((student) => student.user.id === studentId)} />))
+                        }
+                    </Grid>
+                    <Grid alignItems={parentIsLoggedIn && "flex-end"} container
+                        direction={parentIsLoggedIn ? "column" : "row"}
+                        item
+                        justify={parentIsLoggedIn ? "flex-end" : "space-between"}
+                        spacing={parentIsLoggedIn && 4}
+                        style={{"marginTop": "50px"}}>
+                        {parentIsLoggedIn ?
```
Comment (dhuangg) [inline src/components/FeatureViews/Registration/RegistrationCart/RegistrationCartContainer.js:250]: you can remove the parentIsLoggedIn logic - parents should be allowed to pay through stripe

File: src/components/FeatureViews/Registration/RegistrationCart/RegistrationCartContainer.js [hunk 2]
```diff
@@ -67,284 +67,285 @@ const GET_COURSES_AND_STUDENTS_TO_REGISTER = gql`
...
+                            <>
+                                <Grid item xs={5}>
```
Comment (dhuangg) [inline src/components/FeatureViews/Registration/RegistrationCart/RegistrationCartContainer.js:252]: If you remove the parent access control, you can refactor the code from line 252 - 276 to handle the pay later flow instead

File: src/components/FeatureViews/Registration/RegistrationCart/PaymentBoard.js [hunk 3]
```diff
@@ -60,400 +64,479 @@ const CREATE_ENROLLMENTS = gql`mutation CreateEnrollments($enrollments:[Enrollme
...
+            </Grid>
+            <Grid container item justify="flex-end">
+                <Grid item>
+                    <ResponsiveButton
+                        color="primary"
+                        data-cy="pay-action"
+                        disabled={paymentMethod === null || priceQuote.total === "-" || priceQuote < 0}
+                        onClick={handlePayment}
+                        variant="contained">
+                        {paymentMethod === "credit_card" ? "Pay Now" : "Pay Later"}
```
Comment (dhuangg) [inline src/components/FeatureViews/Registration/RegistrationCart/PaymentBoard.js:534]: this doesn't follow Gaby's design - we're supposed to enable/disable buttons depending on payment methods. not completely switch them out. Also I'm assuming the pay later flow hasn't been implemented?
Screen Shot 2020-12-13 at 2 20 58 PM

Screen Shot 2020-12-13 at 2 21 12 PM

**Review by dhuangg — APPROVED** [review]
everything makes sense, but looks like this is just a PR for the pay now flow. we'll need another one for the pay later flow

File: src/components/FeatureViews/Registration/RegistrationCart/PaymentBoard.js (hunk 3 above)
Comment (NeilAllavarpu) [reply src/components/FeatureViews/Registration/RegistrationCart/PaymentBoard.js:534]: yeah, i havent gotten around to pay later yet

**Pull request merged**